test/e2e/scripts/setup_maas_users_openshift.sh

```diff
@@ -0,0 +1,137 @@
+# Does NOT remove or modify existing identity providers/users
+# Assumes 'htpasswd' is already installed.
+# ==========================================================
+
+set -euo pipefail
+
+# --- Configuration ---
+HTPASSWD_FILE="./maas-users.htpasswd"
+HTPASSWD_SECRET_NAME="maas-htpasswd-secret"
+IDP_NAME="maas-htpasswd-provider"
+
+# Openshift users
+OPENSHIFT_ADMIN_USER="maas-admin-user"
+OPENSHIFT_DEV_USER="maas-dev-user"
+
+# Passwords
+OPENSHIFT_ADMIN_PASS="${MAAS_ADMIN_PASSWORD:-$(openssl rand -hex 12)}"
+OPENSHIFT_DEV_PASS="${MAAS_DEV_PASSWORD:-$(openssl rand -hex 12)}"
+
+export OPENSHIFT_ADMIN_USER
+export OPENSHIFT_DEV_USER
+export OPENSHIFT_ADMIN_PASS
+export OPENSHIFT_DEV_PASS
+
+echo "=== Setting up MAAS HTPasswd Identity Provider ==="
+
+# --- Create htpasswd file ---
+echo "Creating htpasswd file with users..."
+htpasswd -c -B -b "${HTPASSWD_FILE}" "${OPENSHIFT_ADMIN_USER}" "${OPENSHIFT_ADMIN_PASS}"
+htpasswd -B -b "${HTPASSWD_FILE}" "${OPENSHIFT_DEV_USER}" "${OPENSHIFT_DEV_PASS}"
+
+# --- Create or update secret ---
+echo "Creating/Updating secret..."
+oc create secret generic "${HTPASSWD_SECRET_NAME}" \
+  --from-file=htpasswd="${HTPASSWD_FILE}" \
+  -n openshift-config --dry-run=client -o yaml | oc apply -f -
+
+# --- Create or patch OAuth configuration ---
+if ! oc get oauth cluster &>/dev/null; then
+  echo "No existing OAuth found. Creating a new one..."
+  cat <<EOF | oc apply -f -
+apiVersion: config.openshift.io/v1
+kind: OAuth
+metadata:
+  name: cluster
+spec:
+  identityProviders:
+  - name: ${IDP_NAME}
+    mappingMethod: claim
+    type: HTPasswd
+    htpasswd:
+      fileData:
+        name: ${HTPASSWD_SECRET_NAME}
+EOF
+else
+  echo "OAuth exists. Checking if MAAS HTPasswd provider already exists..."
+  
+  # Check if our identity provider already exists
+  if oc get oauth cluster -o jsonpath="{.spec.identityProviders[?(@.name=='${IDP_NAME}')].name}" 2>/dev/null | grep -q "${IDP_NAME}"; then
+    echo "MAAS HTPasswd provider already exists, skipping..."
+  else
+    echo "Adding new MAAS HTPasswd provider..."
+    # Create a simple patch file to avoid YAML escaping issues
+    cat > /tmp/oauth-patch.json <<JSONEOF
+{
+  "spec": {
+    "identityProviders": [
+      {
+        "name": "${IDP_NAME}",
+        "mappingMethod": "claim",
+        "type": "HTPasswd",
+        "htpasswd": {
+          "fileData": {
+            "name": "${HTPASSWD_SECRET_NAME}"
+          }
+        }
+      }
+    ]
+  }
+}
+JSONEOF
+    
+    # Apply the patch from file
+    if oc patch oauth cluster --type=merge --patch-file /tmp/oauth-patch.json; then
+      echo "Successfully added MAAS HTPasswd provider"
+    else
+      echo "Patch failed, but continuing..."
+    fi
+    
+    # Cleanup
+    rm -f /tmp/oauth-patch.json
+  fi
+fi
+
+# --- Wait for rollout ---
+echo "Waiting for authentication rollout..."
+sleep 5
+oc rollout status deployment/oauth-openshift -n openshift-authentication --timeout=180s || true
```

Comment:
Maybe we can wait for the condition here instead?
```suggestion
oc -n openshift-authentication wait deployment/oauth-openshift --for=condition=Available --timeout=5m
```